src/post/Write/DraftRow.js

```diff
@@ -34,14 +31,23 @@ class DraftRow extends React.Component {
       <div>
         <Link to={{ pathname: '/write', search: `?draft=${id}` }}>
           <h3>
-            {draftTitle.length === 0 ? <FormattedMessage id="draft_untitled" defaultMessage="Untitled draft" /> : draftTitle}
+            {draftTitle.length === 0 ? (
+              <FormattedMessage id="draft_untitled" defaultMessage="Untitled draft" />
+            ) : (
+              draftTitle
+            )}
           </h3>
         </Link>
-        <div>
-          <Button loading={pending} type="danger" onClick={this.handleDeleteClick}>
-            <FormattedMessage id="draft_delete" defaultMessage="Delete this draft" />
-          </Button>
-        </div>
+        <Button type="danger" onClick={this.showModal}>
+          <FormattedMessage id="draft_delete" defaultMessage="Delete this draft" />
+        </Button>
+        {this.state.showModalDelete && (
+          <DeleteDraftModal
+            draftId={this.props.id}
+            onDelete={this.hideModal}
```

Comment:
We don't need to have `onDelete` handler there.
`hideModal` uses `this.setState` which is no-op on unmounted components. `DeleteDraftModal` deletes draft so `DraftRow` is unmounted before `hideModal` fires, causing this error:
```
Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the DraftRow component.
```